src/main/java/edu/cornell/kfs/tax/batch/service/impl/TransactionDetailGeneratorBase.java

```diff
@@ -0,0 +1,278 @@
+
+    protected TransactionDetailGeneratorBase(final TaxBatchConfig config, final TaxDtoRowMapper<T> rowMapper,
+            final TransactionDetailBuilderHelperService helperService, final String taxSourceType,
+            final TaxStatType... initialZeroValueStats) {
+        Objects.requireNonNull(config, "config cannot be null");
+        Objects.requireNonNull(rowMapper, "rowMapper cannot be null");
+        Objects.requireNonNull(helperService, "helperService cannot be null");
+        Validate.notBlank(taxSourceType, "taxSourceType cannot be blank");
+        Objects.requireNonNull(initialZeroValueStats, "initialZeroValueStats var-args cannot be null");
+        this.config = config;
+        this.rowMapper = rowMapper;
+        this.helperService = helperService;
+        this.taxSourceType = taxSourceType;
+        this.statistics = new TaxStatistics(initialZeroValueStats);
+        this.pendingBatchInserts = new ArrayList<>(MAX_BATCH_INSERT_SIZE);
+        this.emptyRouteHeader = new RouteHeaderLite();
+        this.numTransactionDetailsInserted = 0;
+    }
+
+    public TaxStatistics generateAndInsertTransactionDetails() throws SQLException {
+        LOG.info("generateAndInsertTransactionDetails, Starting generation of {} transaction details", taxSourceType);
+        while (rowMapper.moveToNextRow()) {
+            final TransactionDetail detail = generateTransactionDetailFromCurrentSourceDataRow();
+            pendingBatchInserts.add(detail);
+            if (pendingBatchInserts.size() >= MAX_BATCH_INSERT_SIZE) {
+                performBatchInserts();
+            }
+        }
+
+        if (pendingBatchInserts.size() > 0) {
+            performBatchInserts();
+        }
+
+        LOG.info("generateAndInsertTransactionDetails, Finished generation of {} {} transaction details",
+                numTransactionDetailsInserted, taxSourceType);
+
+        return statistics;
+    }
+
+    protected void performBatchInserts() {
+        LOG.info("performBatchInserts, Preparing to insert next batch of {} {} transaction details",
+                pendingBatchInserts.size(), taxSourceType);
+        final List<String> documentIds = getDocumentIdsFromPendingBatchInserts();
+        final List<RouteHeaderLite> routeHeaders = helperService.getBasicRouteHeaderData(documentIds);
+        final Map<String, RouteHeaderLite> routeHeadersMap = routeHeaders.stream()
+                .collect(Collectors.toUnmodifiableMap(RouteHeaderLite::getDocumentNumber, Function.identity()));
+
+        final List<TransactionDetail> detailsToInsert = pendingBatchInserts.stream()
+                .map(detail -> prepareTransactionDetailForInsertionIfPossible(detail, routeHeadersMap))
+                .filter(ObjectUtils::isNotNull)
+                .collect(Collectors.toUnmodifiableList());
+
+        if (detailsToInsert.isEmpty()) {
+            LOG.warn("performBatchInserts, None of the batch's {} {} transaction details are eligible for insertion",
```

Comment:
Are either of these two "warning" conditions serious issues that will need to be more visible so something can be corrected?